### PR TITLE
extend tracked file size only if current write request crossed assumed file end

### PR DESCRIPTION
**Pull request by bkmit**

### CODE REVIEW

**Comment by Castaglia** [comment]
Can you provide a description of why this is useful?  What's your use case?  What's the desired functionality, and why?  Thanks!

**Comment by bkmit** [comment]
1. Possible negative value indicating write error wrongly used for size calculating.
2. SFTP protocol allow writing in random region of existing file not always expanding it. This patch allow properly calculate file size in this case. This calculated value is used for enforcing file size limit.

**Comment by mfuterko** [comment]
I've extended this pull request with additional flags for READ/WRITE request to indicate whether the dir_chack() functions were called on the file handle. There are more similar places for optimisation.

File: contrib/mod_sftp/fxp.c [hunk 1]
```diff
@@ -11227,7 +11237,8 @@ static int fxp_handle_write(struct fxp_packet *fxp) {
    * want to track the on-disk size for enforcing limits such as
    * MaxStoreFileSize.
    */
-  fxh->fh_st->st_size += res;
+  if (res > 0 && (offset + res) > fxh->fh_st->st_size)
```
Comment (Castaglia) [inline contrib/mod_sftp/fxp.c]: This logic has been added to master; thanks!

**Pull request closed without merging**